server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -816,6 +822,122 @@ private ClusterMetadataManifest uploadManifest(
         }
     }
 
+    private ClusterState readClusterStateInParallel(
+        ClusterState previousState,
+        ClusterMetadataManifest manifest,
+        String clusterName,
+        String clusterUUID,
+        String localNodeId,
+        List<UploadedIndexMetadata> indicesToRead,
+        Map<String, UploadedMetadataAttribute> customToRead,
+        boolean readCoordinationMetadata,
+        boolean readSettingsMetadata,
+        boolean readTransientSettingsMetadata,
+        boolean readTemplatesMetadata,
+        boolean readDiscoveryNodes,
+        boolean readClusterBlocks,
+        List<UploadedIndexMetadata> indicesRoutingToRead,
```

Comment:
How are we getting this list `indicesRoutingToRead`

Reply:
Added.